### PR TITLE
Update head.py

### PR DESCRIPTION
**Pull request by alanyee**
Replace scalar_summary for tf.summary.scalar

### CODE REVIEW

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Review by rmlarsen — APPROVED** [review]

**Comment by alanyee** [comment]
#11159

File: tensorflow/contrib/learn/python/learn/estimators/head.py [hunk 1]
```diff
@@ -35,8 +35,6 @@
 from tensorflow.python.framework import ops
 from tensorflow.python.framework import sparse_tensor
 from tensorflow.python.ops import array_ops
-from tensorflow.python.ops import control_flow_ops
```
Comment (rmlarsen) [inline tensorflow/contrib/learn/python/learn/estimators/head.py]: Oops! It looks like this import is still needed.

**Review by rmlarsen — CHANGES_REQUESTED** [review]

File: tensorflow/contrib/learn/python/learn/estimators/head.py (hunk 1 above)
Comment (alanyee) [inline tensorflow/contrib/learn/python/learn/estimators/head.py]: That's interesting. Why is it still needed?

Comment (alanyee) [inline tensorflow/contrib/learn/python/learn/estimators/head.py]: Oh wait, never mind. I see that control_flow_ops is actually been getting called elsewhere in the code. I must have been looking at the different code file when I ran my search.

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Comment by rmlarsen** [comment]
@alanyee lease address the remaining test failures in:

https://ci.tensorflow.org/job/tensorflow-pull-requests-cpu/6219/consoleFull

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Review by rmlarsen — DISMISSED** [review]

**Comment by rmlarsen** [comment]
@alanyee @ispirmustafa It looks like there are some non-trivial changes in the tags causing a failure in the unit tests:

======================================================================
FAIL: testBinaryClassificationEval (__main__.BinaryClassificationHeadTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/head_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/head_test.py", line 814, in testBinaryClassificationEval
    _assert_summary_tags(self, ["loss"])
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/head_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/head_test.py", line 90, in _assert_summary_tags
    test_case.assertItemsEqual(expected_tags or [], actual_tags)
AssertionError: Element counts were not equal:
First has 1, Second has 0:  'loss'
First has 0, Second has 1:  u'binary_logistic_head/loss'

**Comment by alanyee** [comment]
Ah, I have been trying to get around that problem which has occurred for another PR: 
```----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/head_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/head_test.py", line 1444, in testMultiClassWithLabelKeysEvalAccuracy1
    _assert_summary_tags(self, ["loss"])
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/head_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/head_test.py", line 90, in _assert_summary_tags
    test_case.assertItemsEqual(expected_tags or [], actual_tags)
AssertionError: Element counts were not equal:
First has 1, Second has 0:  'loss'
First has 0, Second has 1:  u'multi_class_head/loss'
```

The problem is essentially that the training and eval losses show up in different graphs which is why the deprecated API was used in the first place.

**Comment by ispirmustafa** [comment]
Yes we intentionally keep deprecated summary API in head.py under contrib.
The heads we moved to core (python/estimator/canned/head.py) we use the new summary API.
Closing this since we'll not change deprecated api usages in contrib version of head.

**Comment by alanyee** [comment]
@ispirmustafa okay but i wanted to test out one more change to see if it would work

**Pull request closed without merging**